src/backend/cdb/cdbmetadatacache.c

```diff
@@ -619,6 +619,9 @@ BlockLocation *GetHdfsFileBlockLocations(const HdfsFileInfo *file_info, uint64_t
             if (cache_entry->block_num <= 1)
             {
                 // only one file, re-fetch 
+                LWLockRelease(MetadataCacheLock); 
+                
+                LWLockAcquire(MetadataCacheLock, LW_EXCLUSIVE);
```

Comment:
Can this create deadlock situations? Upgrading locks usually is a tricky situation, can lead to a situation where:
- proc A has SHARED lock on foo and bar
- proc B has SHARED lock on foo and bar

- proc A requests EXCLUSIVE lock on foo and gets it
- proc B requests EXCLUSIVE lock on bar and gets it
- proc A requests EXCLUSIVE lock on bar and waits
- proc B requests EXCLUSIVE lock on foo and waits - deadlock